app/policies/mixins/rbac_mixin.rb

```diff
@@ -0,0 +1,106 @@
+module Mixins
+  module RBACMixin
```

Comment:
The biggest problem I have with this being a mixin is that this contains *a lot* of logic. When you mix this into another class, you're adding all of that functionality and that makes it really difficult to test.

I know we're trying to get this PR through rather quickly, so it should probably be handled in a separate PR, but I would like to see this mixin broken out into a separate class and then a corresponding spec written against it. I would imagine you'll find it pretty ridiculous the levels of contexts you would need to create (ex: #resource check has easily 10+ levels of cyclomatic complexity). This would further indicate that we could potentially break it up into smaller pieces and have simpler classes to follow.

Reply:
Refactored per discussion with Madhu. The left logic is necessary to be shared with different policys.